public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,223 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, DataSourceSettings, SelectableValue } from '@grafana/data';
+import { DataSourceHttpSettings, FormLabel, Input, SecretFormField, Select } from '@grafana/ui';
+import { InfluxOptions, InfluxSecureJsonData } from '../types';
+
+export type Props = DataSourcePluginOptionsEditorProps<InfluxOptions>;
+
+type InfluxDataSourceSettings = DataSourceSettings<InfluxOptions, InfluxSecureJsonData>;
+
+export interface State {
+  config: InfluxDataSourceSettings;
+  httpModes: SelectableValue[];
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { options } = this.props;
+
+    this.state = {
+      config: ConfigEditor.defaults(options),
+      httpModes: [{ label: 'GET', value: 'GET' }, { label: 'POST', value: 'POST' }],
+    };
+  }
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    return {
+      ...state,
+      config: ConfigEditor.defaults(props.options),
+    };
+  }
+
+  static defaults = (options: any) => {
```

Comment:
is this `any` or `InfluxDataSourceSettings`?

Reply:
`InfluxDataSourceSettings`. It's actually any on azure and cloudwatch too. Will need to update there as well.